src/NLog/Common/ConversionHelpers.cs

```diff
@@ -47,38 +47,53 @@ public static class ConversionHelpers
         /// <param name="inputValue">Input value</param>
         /// <param name="resultValue">Output value</param>
         /// <param name="defaultValue">Default value</param>
-        /// <returns>Returns failure if the input value could not be parsed</returns>
+        /// <returns>Returns <paramref name="defaultValue"/> if the input value could not be parsed</returns>
         public static bool TryParseEnum<TEnum>(string inputValue, out TEnum resultValue, TEnum defaultValue = default(TEnum)) where TEnum : struct
         {
-            if (string.IsNullOrEmpty(inputValue))
+            if (!TryParseEnum(inputValue, true, out resultValue))
             {
                 resultValue = defaultValue;
-                return true;
+                return false;
             }
+            return true;
+        }
 
-            if (!TryParse(inputValue, true, out resultValue))
+        /// <summary>
+        /// Converts input string value into <see cref="System.Enum"/>. Parsing is case-insensitive.
+        /// </summary>
+        /// <param name="inputValue">Input value</param>
+        /// <param name="enumType">The type of the enum</param>
+        /// <param name="resultValue">Output value. Null if parse failed</param>
+        internal static bool TryParseEnum(string inputValue, Type enumType, out object resultValue)
+        {
+            // Note: .NET Standard 2.1 added a public Enum.TryParse(Type)
```

Comment:
Maybe add `if (StringHelpers.IsNullOrWhiteSpace(inputValue)) return false` since it is a common scenario.